source/common/stats/allocator_impl.h

```diff
@@ -93,6 +103,15 @@ class AllocatorImpl : public Allocator {
   std::vector<GaugeSharedPtr> deleted_gauges_ ABSL_GUARDED_BY(mutex_);
   std::vector<TextReadoutSharedPtr> deleted_text_readouts_ ABSL_GUARDED_BY(mutex_);
 
+  template <typename StatType>
+  using StatPointerSet = absl::flat_hash_set<StatType*, absl::Hash<StatType*>>;
```

Comment:
does it work to omit the hashing type here? I would've thought the default would be fine.

Normally I think it's best practice if you are specifying a hasher to also specify the comparator, as it's usually bad news if they don't match semantically.